--- a/rust/private/rustfmt.bzl
+++ b/rust/private/rustfmt.bzl
@@ -67 +67 @@
-def _perform_check(edition, srcs, ctx):
+def _perform_check(edition, srcs, root, ctx):
@@ -79 +78,0 @@
-    args.add("--config", "skip_children=true")
@@ -81 +80 @@
-    args.add_all(srcs)
+    args.add(root)
@@ -144 +143 @@
-    marker = _perform_check(info.edition, info.srcs, ctx)
+    marker = _perform_check(info.edition, info.srcs, _get_rustfmt_ready_crate_info(target).root, ctx)
