--- a/rust/private/rustfmt.bzl
+++ b/rust/private/rustfmt.bzl
@@ -64,7 +64,7 @@
 
     return srcs
 
-def _perform_check(edition, srcs, ctx):
+def _perform_check(edition, srcs, root, ctx):
     rustfmt_toolchain = ctx.toolchains[Label("//rust/rustfmt:toolchain_type")]
 
     config = ctx.file._config
@@ -76,9 +76,8 @@
     args.add(rustfmt_toolchain.rustfmt)
     args.add("--config-path", config)
     args.add("--edition", edition)
-    args.add("--config", "skip_children=true")
     args.add("--check")
-    args.add_all(srcs)
+    args.add(root)
 
     ctx.actions.run(
         executable = ctx.executable._process_wrapper,
@@ -141,7 +140,7 @@
     if not info.srcs:
         return []
 
-    marker = _perform_check(info.edition, info.srcs, ctx)
+    marker = _perform_check(info.edition, info.srcs, _get_rustfmt_ready_crate_info(target).root, ctx)
 
     return [
         OutputGroupInfo(
